From 9a335e1d316572aeff4632047dce1600825cbfba Mon Sep 17 00:00:00 2001
From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Fri, 31 May 2024 13:00:27 +0200
Subject: [PATCH 1/3] cmake: Detect armv7l as ARM architecture

On Ubuntu armhf:

```
$ uname -p
armv7l
```

So detect `armv7l` as ARM architecture as well.
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: procdump/CMakeLists.txt
===================================================================
--- procdump.orig/CMakeLists.txt
+++ procdump/CMakeLists.txt
@@ -113,7 +113,7 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux"
     set(CLRHOSTDEF -DHOST_X86)
   elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL armv6 OR CMAKE_SYSTEM_PROCESSOR STREQUAL armv6l)
     set(CLRHOSTDEF -DHOST_ARM -DHOST_ARMV6)
-  elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL arm OR CMAKE_SYSTEM_PROCESSOR STREQUAL armv7-a)
+  elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL arm OR CMAKE_SYSTEM_PROCESSOR STREQUAL armv7-a OR CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l OR CMAKE_SYSTEM_PROCESSOR STREQUAL armv8l)
     set(CLRHOSTDEF -DHOST_ARM)
   elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64 OR CMAKE_SYSTEM_PROCESSOR STREQUAL arm64)
     set(CLRHOSTDEF -DHOST_ARM64 -DHOST_64BIT)
Index: procdump/profiler/inc/pal_mstypes.h
===================================================================
--- procdump.orig/profiler/inc/pal_mstypes.h
+++ procdump/profiler/inc/pal_mstypes.h
@@ -503,7 +503,8 @@ UShortToPtr(
 typedef _W64 __int32 INT_PTR;
 typedef _W64 unsigned __int32 UINT_PTR;
 
-typedef _W64 __int32 LONG_PTR;
+typedef _W64 __int32 LONG_PTR, *PLONG_PTR;
+
 typedef _W64 unsigned __int32 ULONG_PTR, *PULONG_PTR;
 typedef _W64 unsigned __int32 DWORD_PTR, *PDWORD_PTR;
 
